runtime-modules/bureaucracy/src/lib.rs

```diff
@@ -835,6 +856,93 @@ decl_module! {
             // Trigger event
             Self::deposit_event(RawEvent::WorkerOpeningFilled(worker_opening_id, worker_application_id_to_worker_id));
         }
+
+        // ****************** Worker stakes **********************
+
+        /// Slashes the worker stake, demands a leader origin. No limits, no actions on zero stake.
+        /// If slashing balance greater than the existing stake - stake is slashed to zero.
+        pub fn slash_worker_stake(origin, worker_id: WorkerId<T>, balance: BalanceOf<T>) {
+            Self::ensure_origin_is_set_lead(origin)?;
+
+            let worker = Self::ensure_worker_exists(&worker_id)?;
+
+            ensure!(balance != <BalanceOf<T>>::zero(), Error::StakeBalanceCannotBeZero);
+
+            let stake_profile = worker.role_stake_profile.ok_or(Error::NoWorkerStakeProfile)?;
+
+            //
+            // == MUTATION SAFE ==
+            //
+
+            ensure_on_wrapped_error!(
```

Comment:
I think this needs to be above the `MUTATION SAFE`

Reply:
Why? slash_immediate() - will cause changes in the storage. And it should be after the 'mutation' sign.

Reply:
Because mutation is not safe on line 876 at the moment, and this gives that impression, which could lead someone to add other business logic in that wedge. Its only safe _after_  `slash_immediate` has successfully executed, because it checks the very last condition internally.